frontend/src/metabase/selectors/app.ts

```diff
@@ -49,6 +49,76 @@ export const getIsAdminApp = createSelector([getRouterPath], path => {
   return path.startsWith("/admin/");
 });
 
+export const getIsCollectionPathVisible = createSelector(
+  [getQuestion, getDashboard, getRouterPath, getIsEmbedded, getEmbedOptions],
+  (question, dashboard, path, isEmbedded, embedOptions) => {
+    if (isEmbedded && !embedOptions.breadcrumbs) {
+      return false;
+    }
+
+    return (
+      ((question != null && question.isSaved()) || dashboard != null) &&
+      PATHS_WITH_COLLECTION_BREADCRUMBS.some(pattern => pattern.test(path))
+    );
+  },
+);
+
+export const getIsQuestionLineageVisible = createSelector(
+  [getQuestion, getOriginalQuestion, getRouterPath],
+  (question, originalQuestion, path) =>
+    question != null &&
+    !question.isSaved() &&
+    originalQuestion != null &&
+    !originalQuestion.isDataset() &&
+    PATHS_WITH_QUESTION_LINEAGE.some(pattern => pattern.test(path)),
+);
+
+export const getIsNavBarEnabled = createSelector(
+  [
+    getUser,
+    getRouterPath,
+    getIsEditingDashboard,
+    getIsEmbedded,
+    getEmbedOptions,
+  ],
+  (currentUser, path, isEditingDashboard, isEmbedded, embedOptions) => {
+    if (!currentUser || isEditingDashboard) {
+      return false;
+    }
+    if (isEmbedded && !embedOptions.side_nav) {
+      return false;
+    }
+    if (isEmbedded && embedOptions.side_nav === "default") {
+      return EMBEDDED_PATHS_WITH_NAVBAR.some(pattern => pattern.test(path));
+    }
+    return !PATHS_WITHOUT_NAVBAR.some(pattern => pattern.test(path));
+  },
+);
+
+const getIsEmbeddedAppBarVisible = createSelector(
+  [
+    getEmbedOptions,
+    getIsQuestionLineageVisible,
+    getIsCollectionPathVisible,
+    getIsNavBarEnabled,
+  ],
+  (
+    embedOptions,
+    isQuestionLineageVisible,
+    isCollectionPathVisible,
+    isNavBarEnabled,
+  ) => {
+    const anyEmbeddedAppBarElementVisible =
+      isNavBarEnabled ||
+      embedOptions.search ||
+      embedOptions.new_button ||
+      embedOptions.logo ||
+      isQuestionLineageVisible ||
+      isCollectionPathVisible;
```

Comment:
These are all the flags that control what element will be displayed in the app bar in https://github.com/metabase/metabase/blob/master/frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx.

[So if nothing is displayed we can remove the app bar out entirely.](https://metaboat.slack.com/archives/C045W5MF84U/p1672853105921429?thread_ts=1672847315.846639&cid=C045W5MF84U)